src/DiffPatcher.js

```diff
@@ -1,11 +1,27 @@
 import { DiffPatcher } from 'jsondiffpatch/src/diffpatcher';
 
+const isLoop = function(context) {
+  // Based on https://github.com/benjamine/jsondiffpatch/issues/152
+  let i=0;
+  let parent = context.parent;
+  while (parent !== undefined) {
+    if (parent.left === context.left && parent.right === context.right) {
+      return true;
+    }
+    parent = parent.parent;
+    i++;
+    if (i>10) return false; // We don't want to go that deep
```

Comment:
It still hang on circular references which are deeper than 10th level, but it shouldn't be the case for Redux, where the data would be normalized. Though we could `return true` here and solve such cases and also forbid comparing deep nested objects with more than 10 levels.